compiler/src/iree/compiler/Codegen/Common/GPU/GPUHeuristics.cpp

```diff
@@ -478,26 +485,78 @@ sortMMAIntrinsics(GPUMatmulShapeType problem,
   return sortedIntrinsics;
 }
 
-FailureOr<GPUMMASchedule> deduceMMASchedule(
-    const GPUMatmulShapeType &problem, ArrayRef<GPUIntrinsicType> intrinsics,
-    const GPUMMAHeuristicSeeds &seeds, int64_t sharedMemLimitInBytes,
-    int64_t subgroupSize, bool transposedLhs, bool transposedRhs,
-    bool canUpcastAcc, bool mustBeAligned, bool doCPromotion) {
-
-  sortMMAIntrinsics(problem, intrinsics);
+static int64_t adjustSeedsForWgpCount(const GPUMatmulShapeType &problem,
+                                      const GPUIntrinsicType &intrinsic,
+                                      int64_t wgpCount,
+                                      int64_t bestSubgroupCountPerWorkgroup,
+                                      int64_t bestMNTileCountPerSubgroup) {
+  int64_t mSize = ShapedType::getNumElements(problem.mSizes);
+  int64_t nSize = ShapedType::getNumElements(problem.nSizes);
+  int64_t kSize = ShapedType::getNumElements(problem.kSizes);
+  float arithmeticIntensity =
+      (2.0f * mSize * nSize * kSize) /
+      static_cast<float>(mSize * nSize + nSize * kSize + mSize * kSize);
+
+  // TODO(jerryyin): compute arithmetic intensity bound based on the information
+  // from the target chip.
+  if (arithmeticIntensity <= 10.0f) {
+    LDBG() << "Arithmetic intensity is too low, " << arithmeticIntensity
+           << ", skipping adjustment of seeds for workgroup count.";
+    return bestMNTileCountPerSubgroup;
+  }
+  auto computeWorkgroupCount = [&]() {
```

Comment:
nit: you don't need empty parens
```suggestion
  auto computeWorkgroupCount = [&] {
```